engine/swarm/store_sensitive_strings.md

```diff
@@ -0,0 +1,234 @@
+---
+title: Manage sensitive strings (secrets) for Docker services
+description: How to securely store, retrieve, and use sensitive strings (secrets) in Docker services
+keywords: swarm, secrets, credentials, sensitive strings, security, encryption, encryption at rest
+---
+
+## About secrets
+
+In terms of Docker Swarm services, a _secret_ is a string, such as a password,
+credit card number, SSH private key, SSL certificate, or another piece of
+sensitive data that should not be transmitted over a network or stored in a
+Dockerfile or in your application's source code in plain text. Starting in
+Docker 1.13, Docker can encrypt and manage these secrets for you. Consider the
+following examples when you might want to manage sensitive strings using Docker:
+
+- Your service relies on a password-protected database, and you want to protect
+  the database's credentials.
+- You need a central location to administer and manage credentials needed by
+  different services running in your swarm.
+- Your application relies upon a SSH private key, and you want to protect the
+  key.
+- You run an application, such as a web server, which needs to use a SSL
+  certificate, and you want to protect the certificate itself.
+- You use a different SSL certificate in development and production environments
+  but do not want to change the image or store the SSL certificate in the image.
+- Your application uses infrastructure which is metered and needs a credential,
+  such as Amazon AWS, EC2, or S3, and you want to prevent unauthorized use of
+  this infrastructure.
+- You are a developer building an application which processes sensitive data
+  to which you do not need or want direct access. Your administrator may
+  populate the secret key with a mock value while you are developing the
+  application, and change it to use the real value in production.
+
+In these and many other situations, managing sensitive strings using Docker
+secrets makes sense.
+
+## How Docker manages secrets
+
+When you add a secret to the swarm, Docker sends the secret to each manager
+node over a network connection which uses mutual TLS. When a manager receives
+a secret, it encrypts it with an encryption key before writing it to
+the RAFT logs, just as other RAFT log data is encrypted. The encryption key
+itself has already been propagated to the swarm node using mutual TLS.
+
+>**Warning**: RAFT data is encrypted in Docker 1.13 and newer. If some of your
+Swarm nodes run an earlier version, the secrets are stored unecrypted in
+those nodes' RAFT logs.
+
+When you create a service, you can give the service access to one or more
+secrets. You can update a service to grant it access to additional secrets or
+revoke its access to a given secret at any time.
+
+The Engine managing a given worker node creates a bind mount within
+`/run/secrets/` on the worker node for each secret. From the point of view of
```

Comment:
@ehazlett `/run/secrets` runs afoul of FHS standards. Do we link to `/var/run` to `/run`?

From https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard:

> In FHS 3.0, /var/run is replaced by /run; a system should either continue to provide a /var/run directory, or provide a symbolic link from /var/run to /run, for backwards compatibility.[10]

I think we just need to make sure that `/run` is not a fork of `/var/run`. Accessing `/var/run/secrets` or `/run/secrets` should be equivalent, recommending the later.